### PR TITLE
[Torch Dialect] Support recompose aten.split.Tensor + prim.ListUnpack

### PR DESCRIPTION
**Pull request by qingyunqu**

### CODE REVIEW

File: python/torch_mlir_e2e_test/test_suite/slice_like.py [hunk 1]
```diff
@@ -592,15 +592,34 @@ def __init__(self):
...
         return torch.ops.aten.sub(splits[0], splits[1])
 
 @register_test_case(module_factory=lambda: SplitTensorGetItem_Module())
 def SplitTensorGetItem_Module_basic(module, tu: TestUtils):
-    module.forward(tu.rand(2, 3, 4))
+    module.forward(tu.rand(3, 3, 4))
+
+# ==============================================================================
+
+class SplitTesnorListUnpack_Module(torch.nn.Module):
```
Comment (mgehre-amd) [inline python/torch_mlir_e2e_test/test_suite/slice_like.py:607]:
```suggestion
class SplitTensorListUnpack_Module(torch.nn.Module):
```

Comment (qingyunqu) [reply python/torch_mlir_e2e_test/test_suite/slice_like.py]: Done, thanks!

File: python/torch_mlir_e2e_test/test_suite/slice_like.py [hunk 2]
```diff
@@ -665,15 +665,34 @@ def __init__(self):
     @export
     @annotate_args([
         None,
-        ([2, 3, 4], torch.float32, True),
+        ([3, 3, 4], torch.float32, True),
     ])
     def forward(self, x):
-        splits = torch.ops.aten.split(x, 1, 0)
+        splits = torch.ops.aten.split(x, 2, 0)
```
Comment (ramiro050) [inline python/torch_mlir_e2e_test/test_suite/slice_like.py:671]: What's the reason for this change?

**Review by ramiro050 — APPROVED** [review]
LGTM, just one question

File: python/torch_mlir_e2e_test/test_suite/slice_like.py (hunk 2 above)
Comment (qingyunqu) [reply python/torch_mlir_e2e_test/test_suite/slice_like.py:671]: Just test slice op with end index out of upper bound.
